compose-shared-instance-env.yml

```
@@ -60,10 +63,33 @@ services:
     hostname: api
     container_name: ed-fi-ods-api
 
+  adminapp:
+    image: edfialliance/ods-admin-app:${ODS_ADMIN_APP_TAG}
+    environment:
+      POSTGRES_USER: "${POSTGRES_USER}"
+      POSTGRES_PASSWORD: "${POSTGRES_PASSWORD}"
+      ODS_DB: db-ods
+      ADMIN_DB: db-admin
+      API_MODE: "SharedInstance"
+      ENCRYPTION_KEY: "${ENCRYPTION_KEY}"
+    volumes:
+      - ${LOGS_FOLDER}:/app/logs
+      - adminapp-bulk-hashcache:/app/BulkUploadHashCache
```

Should we map to a known folder location instead of having to dig it out of the docker locations

---

This is not a feature that a user would ever look at, so we've done this deliberately. This is a weird process that writes out a little file with a hash value in it, as a sort of log. I have ticket to change that for writing to the database instead. I was flabbergasted when I saw this process.

---

FWIW this corresponds with the EdFi.LoadTools.dll's Configuration.WorkingDirectory property, so it's not an Admin App concept alone.

---

I am more concerned about that the ability of the files being written. Ideally if you deploy in linux you would map to a folder that you can lock down. Also if you are running the docker files under linux (for example Innive), you cannot write to the docker folder.

---

After looking at this several times, I think I understand the problem: there is a typo, and the volume is not actually being used here.

```suggestion
      - vol-adminapp-bulk-hashcache:/app/BulkUploadHashCache
```

---

No, it was not a typo; "vol" is part of the docker volume name but the docker-compose label for the volume is indeed "adminapp-bulk-hashcache". @jamessmckay I have no idea what your objection is here or how you suggest that we remedy this, given that we're using the exact same technique applied for PostgreSQL databases.

---

good point, we should call it out that some implementations may need to map to a locked down folder in the docs...